src/Util/Message.cc

```diff
@@ -107,13 +109,27 @@ CARTA::AddRequiredTiles Message::AddRequiredTiles(
     return add_required_tiles;
 }
 
-CARTA::Point Message::Point(int x, int y) {
+CARTA::Point Message::Point(float x, float y) {
     CARTA::Point point;
     point.set_x(x);
     point.set_y(y);
     return point;
 }
 
+CARTA::Point Message::Point(const casacore::Vector<casacore::Double>& input, int x_index, int y_index) {
+    CARTA::Point point;
+    point.set_x(input(x_index));
+    point.set_y(input(y_index));
+    return point;
+}
+
+CARTA::Point Message::Point(const std::vector<casacore::Quantity>& input, int x_index, int y_index) {
+    CARTA::Point point;
+    point.set_x(input[x_index].getValue());
+    point.set_y(input[y_index].getValue());
+    return point;
+}
+
```

Comment:
These should be calling the function that takes two floats, not duplicating its internals.